x-pack/plugins/lens/public/indexpattern_datasource/indexpattern_suggestions.ts

```diff
@@ -371,7 +371,7 @@ function createNewLayerWithMetricAggregation(
   indexPattern: IndexPattern,
   field: IndexPatternField
 ): IndexPatternLayer {
-  const dateField = indexPattern.fields.find((f) => f.name === indexPattern.timeFieldName)!;
+  const dateField = indexPattern.getFieldByName(indexPattern.timeFieldName!);
```

Comment:
Isn't there an additional `!` necessary to make `dateField` always defined to be identical to the old code?

Reply:
It falls into the `Record<string, ...>` discussed before. In this case TS believes `dataField` is always defined.

Reply:
I thought `getFieldByName` would return `IndexPatternField | undefined`?

Reply:
`field` is an optional argument in the `buildColumn` function:
```js
export function buildColumn({
  op,
  columns,
  field,
  layerId,
  indexPattern,
  suggestedPriority,
  previousColumn,
}: {
  op: OperationType;
  columns: Partial<Record<string, IndexPatternColumn>>;
  suggestedPriority: DimensionPriority | undefined;
  layerId: string;
  indexPattern: IndexPattern;
  field?: IndexPatternField;
  previousColumn?: IndexPatternColumn;
}): IndexPatternColumn 
```

Reply:
I see, thanks